### ISSUE TITLE
Rerender a specific tag/unreleased in the changelog

### ISSUE THREAD

**Issue by ruben-arts** [orhun/git-cliff#433]
**Describe the solution you'd like**

I would like to use `git-cliff` in a GitHub action that runs after every merged PR. Updating the `[unreleased]` section.
My initial instinct was that using the `--prepend` would have this behavior. 

**Describe alternatives you've considered**

I could write some find replace code to manually do the replacement using the output of `git-cliff`. 

**Additional context**

Example of the wokflow I'm looking for:
```markdown
# CHANGELOG

## [1.2.3]
- change by @person in #1234
```
```
git merge feature/x
git-cliff -u --prepend changelog.md
```

```markdown
# CHANGELOG

## [unreleased]
- feature x by @person in #1235

## [1.2.3]
- change by @person in #1234
```

```
git merge feature/y
git-cliff -u --prepend changelog.md
```

```markdown
# CHANGELOG

## [unreleased]
- feature y by @person in #1236
- feature x by @person in #1235

## [1.2.3]
- change by @person in #1234
```

**Comment by orhun** [comment]
Also reported in #544

**Comment by 2bndy5** [comment]
It might be a good idea to require HTML comments as markers between versions (in the tera template). That way there's some guarantee that the CHANGELOG being updated is compatible with this feature.

**Comment by orhun** [comment]
@2bndy5 yup, good idea. I agree that using markers is the safest way to go with this, maybe even instead of parsing markdown.

**Comment by dslemusp** [comment]
Is someone working on this?. I am trying to see if there is a PR open with this?

**Comment by orhun** [comment]
Nope, this isn't implemented yet

**Comment by hgolay** [comment]
Hello, I'm also interested by this feature, is there any plan for it ?

**Comment by ognis1205** [comment]
@hgolay cc: @orhun 

If we want to support this feature properly, `--prepend` needs to insert new commits at the correct position, which likely requires CHANGELOG.md to follow some convention like Keep a Changelog.

If we officially support updating the `[unreleased]` section, I think we need to discuss at the spec level:

- Whether to officially support Keep a Changelog format,
- Or introduce a marker-based approach,
- Or define `--prepend` behavior under the assumption that a specific template is used.

This feels like something we should clarify before implementing.

**Comment by orhun** [comment]
Using a customizable marker would be the simplest and most straightfoward solution for now I think. (Since we are supporting custom formats other than markdown, it should be customizable)

Wdyt? @ognis1205

**Comment by ognis1205** [comment]
> Using a customizable marker would be the simplest and most straightfoward solution for now I think. (Since we are supporting custom formats other than markdown, it should be customizable)
> 
> Wdyt? [@ognis1205](https://github.com/ognis1205)

A marker-based approach does seem like the most reasonable direction here 👍 

I haven't fully thought through all the details yet, but intuitively this feels like it should be implemented as a separate option from `--prepend`, something like `--replace`.

Also, it might make sense for `--prepend` and `--replace` to be mutually exclusive, since they imply fundamentally different behaviors.

Alternatively, although it would introduce a breaking change to the CLI, another possible direction could be to unify this under something like `--mode prepend|replace`.

**Comment by orhun** [comment]
Yeah, `--replace` argument sounds good to me. We should probably have a new value such as `[changelog.replace_marker]` in the configuration file.
